### DISCUSSION TITLE
Unit tests for custom Sphinx extensions, themes, and directives

### DISCUSSION THREAD

**Discussion by kaycebasques** [sphinx-doc/sphinx#12428] [General]
Have you figured out a really solid unit test strategy for your custom Sphinx extension, theme, or directive? Please share your wisdom :)

**Comment by chrisjsewell** [comment]
Heya, well you can have a look at https://github.com/executablebooks/MyST-Parser/tree/master/tests/test_sphinx,
or https://github.com/useblocks/sphinx-needs/tree/master/tests, or anything in https://github.com/sphinx-extensions2

Also https://github.com/sphinx-extensions2/sphinx-pytest

When/if I have time I would eventually like to improve the documentation here on doing this
